### PR TITLE
components/mdns: use semaphore instead of task notification bits (IDFGH-9380) (IDFGH-9388)

### PR DESCRIPTION
**Pull request by boborjan2**
PR for https://github.com/espressif/esp-idf/issues/10754

Component should not alter the task notification bits of the calling task.

### CODE REVIEW

**Review by david-cermak — APPROVED** [review]
Thanks for the quick fix!

File: components/mdns/private_include/mdns_private.h [hunk 1]
```diff
@@ -390,7 +391,6 @@ typedef struct {
     union {
         struct {
             char *hostname;
-            TaskHandle_t calling_task;
```
Comment (david-cermak) [inline components/mdns/private_include/mdns_private.h:393]: Alternatively, we can add the semaphore here
```patch
    struct {
            char *hostname;
-            TaskHandle_t calling_task;
+            SemaphoreHandle_t hostname_ready;
        } hostname_set;
        char *instance;
```

And create it only when setting a new hostname, and delete afterwards  (advantage is that we save a few bytes of heap, when the system is up and running).

Just an optional suggestion to think about. The changes LGTM!

Comment (boborjan2) [reply components/mdns/private_include/mdns_private.h:393]: I would rather not allocate a semaphore in an API call like that. This semaphore might as well be used by other actions in the future(?). Btw, seemingly mdns_server->lock is not used at all.

Comment (david-cermak) [reply components/mdns/private_include/mdns_private.h:393]: > This semaphore might as well be used by other actions in the future(?).

Yes, that's quite possible.

> Btw, seemingly mdns_server->lock is not used at all.

Oh, good point, thanks! This mutex is indeed just allocated but never locked! (probably better idea to save those couple of bytes here!)

**Pull request merged**